### ISSUE TITLE
[Bug Report] ttnn.reciprocal produces incorrect results for BFLOAT8_B

### ISSUE THREAD

**Issue by KalaivaniMCW** [tenstorrent/tt-metal#14672]
**Describe the bug**
ttnn.reciprocal produces incorrect results for `BFLOAT8_B`. 
1. ttnn.reciprocal does not give `inf` when input is 0 but instead gives `1.7014118346046923e+38`
2. When an input value is 0 in `input_tensor(dtype=ttnn.bfloat8_b)`, the output is `1.7014118346046923e+38` and strangely the nearby values become 0 in the output. Please refer attached doc and screenshot below
Test results : [link](https://docs.google.com/spreadsheets/d/108fBlwUwNA3F8sQpz2IHOv6Zrh5uxvPSPlfaxeC3bD4/edit?usp=sharing)

**To Reproduce**

To be tested on a WH machine 
1. checkout kalaivani/recip_bf8b 
2. run `pytest tests/ttnn/unit_tests/operations/eltwise/test_recip.py
` 
or run the code below 
```
import pytest
import torch
import ttnn
from tests.ttnn.utils_for_testing import assert_with_pcc
from tests.ttnn.unit_tests.operations.eltwise.backward.utility_funcs import data_gen_with_range, compare_pcc, data_gen_with_range_dtype

@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
    ),
)
def test_unary_composite_recip_ttnn(input_shapes, device):
    in_data1, input_tensor1 = data_gen_with_range_dtype(input_shapes, -3, 3, device, False, False, ttnn.bfloat8_b)

    output_tensor = ttnn.reciprocal(input_tensor1)
    output_tensor_rm = output_tensor.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch()

    golden_function = ttnn.get_golden_function(ttnn.reciprocal)
    golden_tensor = golden_function(in_data1)

    # for i in range(1):            # Batch size
    #     for j in range(1):        # Channels
    #         for k in range(32):   # Height
    #             for l in range(32):  # Width
    #                 print(f"input: {in_data1[i][j][k][l]} \t tt: {output_tensor_rm[i][j][k][l]} \t torch: {golden_tensor[i][j][k][l]} \n")

    comp_pass = compare_pcc([output_tensor], [golden_tensor], pcc=0.99)
    assert comp_pass
```

**Expected behavior**
The output should match torch result.

**Screenshots**
**dtype=bfloat8_b**
![Image](https://github.com/user-attachments/assets/48b38f62-f9f3-4755-a363-a8a6203d63fc)

**dtype=bfloat16**
![Image](https://github.com/user-attachments/assets/2f4b1200-5792-4124-b498-ddef79bd4a41)

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit)

**Additional context**

**Comment by cmaryanTT** [comment]
It looks like for bfloat8, the values are entirely wrong (always 0, even for obviously controversial inputs). Is that specific to this test configuration, or does the problem occur when a non-controversial input is tested in isolation?

**Comment by ttmtrajkovic** [comment]
Given that the numbers are the same between the 2 tests, this could be the problem of BFP8 quantization.

BFP8 format is format where consecutive 16 numbers share the same exponent, the shared exponent being the exponent of the biggest number. Tests with random numbers from uniform distributions could have consecutive numbers of different magnitude and smaller numbers could just get flushed to zero.
You should check groups of consecutive 16 numbers that include erroneous numbers and check if maximum number may be causing these numbers to get flushed to zero. 
If you're seeing close numbers and still problems, then this is definitely something in the alley of LLKs and possibly HW issues.
Also, please check that outputs of data format const files: `unpack_src/dst_formats.h, pack_src/dst_formats.h` are properly propagating the formats of buffers you wanted.

Milos

**Comment by cmaryanTT** [comment]
Reassigning to @KalaivaniMCW based on Milos' comments. Can you please re-test with groups of 16 repeated inputs, per the comment above.

**Comment by cmaryanTT** [comment]
Deprioritizing accuracy issues per discussion with @ttmtrajkovic , also this is likely expected for the data type

**Comment by ttmtrajkovic** [comment]
@eyonland, @KalaivaniMCW,

The behaviour that you see is expected - `inf` number is absolutely dominating with exponent size in a group of 16, so other numbers are getting flushed to zero.
with block-fp format, 16 numbers share an exponent and it is not possible to treat any number within that group exclusively. The bfp format isn't good to use when nearby numbers could be so different, especially with reciprocal operation which could produce a huge range at the output.

Please handle these exceptions by modeling of output. number-by-number comparison with torch is not the right approach in this case.

Milos

**Comment by eyonland** [comment]
This issue is only impacting backward ops that use bfloat8_b.  My position on this that training will likely not be done with bfloat8_b and so the eltwise ops for backward should NOT support bfloat8_b at all.   I would like to close this issue and deprecate all bfloat_8b dtypes for backward eltwise ops.

**Comment by cmaryanTT** [comment]
I see reciprocal as a generic mathematical op; that's maybe more relevant
to training, but should still support bfloat8_b with a documented caveat.

**Comment by eyonland** [comment]
@dmakoviichuk-tt , what are your thoughts on how we handle bfloat8_b for backward functions given how zero is handled?   Would it be better to continue to provide support for backward ops by doing a typecast to bfloat16 or would it make more sense to completely remove support bfloat8_b entirely on the backward ops?

**Comment by eyonland** [comment]
@mouliraj-mcw , thank you for the draft PR #16422 using where to check when zero then setting the one.   

I'm still thinking that we should not support bfloat_8b for any backward op that depends on the reciprocal for now.  A better solution needs a lot more careful thought and requires input from those that know much more about training.

**Comment by prajaramanTT** [comment]
@KalaivaniMCW is this still an open issue ? If not, can you please close this ? Thanks.

**Comment by eyonland** [comment]
@amahmudTT, we need to update the sweep tests to skip allowing zero in any test that is using bfloat8_b.   Instead we have chosen to document this behavior but leave the capability to use bfloat_8b for now.

Tracking the documentation update under https://github.com/tenstorrent/tt-metal/issues/16350
